### ISSUE TITLE
SyntaxError: Invalid regular expression: invalid group specifier name

### ISSUE THREAD

**Issue by unclekitty** [slevithan/regex#26]
SyntaxError in  `src/subroutines.js`

```javascript
const subroutinePattern = String.raw`\\g<(?<subroutineName>[^>&]+)>`;
const token = new RegExp(String.raw`
${subroutinePattern}
| (?<capturingStart>${capturingDelim})
| \\(?<backrefNum>[1-9]\d*)
| \\k<(?<backrefName>[^>]+)>
| \\?.
`.replace(/\s+/g, ''), 'gsu');
```

Safari browser  Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko

**Comment by slevithan** [comment]
Thanks for the report!

Can you provide more details? I'm not easily able to test Safari.

- What is the full SyntaxError?
- Which version of Safari does that user agent string correspond to?
- Anything else you can think of that might be helpful for reproducing/debugging? E.g. does it reproduce for you on other browsers or computers?

It's not at all obvious why that code would trigger a syntax error, and obviously it is passing the tests in recent versions of Node.js, etc.

**Comment by unclekitty** [comment]
<img width="1000" alt="666dce127fb220f25c631367aca210b5_" src="https://github.com/user-attachments/assets/becdca58-19eb-42f7-8907-02109ad6b1a4">
<img width="1000" alt="aac1c868b30a62da9d4dc7042a4cf4f6_" src="https://github.com/user-attachments/assets/c30fa0bf-143f-4985-820f-6de3f4baae45">

The 'token' value is  ```/\\g<(?<subroutineName>[^>&]+)>|(?<capturingStart>\((?!\?)(?!(?<=\(\?\()DEFINE\))|\(\?<(?![=!])(?<captureName>[^>]+)>)|\\(?<backrefNum>[1-9]\d*)|\\k<(?<backrefName>[^>]+)>|\\?./gsu```,
but in the old version macOS webview, some syntax in the Regex is not supported.

**Comment by slevithan** [comment]
Okay, the screenshot shows the error as "Invalid group specifier name". That helps a little, but unfortunately not very much because there are no group names in the regex that are in fact invalid.

What version of Safari are you running? The `userAgent` string only includes the WebKit version, and I'm not sure how to translate that to a Safari version.

Can you please test in the latest version of Safari (18.1) and see if the problem still occurs?

**Comment by slevithan** [comment]
I found [this mapping](https://github.com/mdn/browser-compat-data/blob/main/browsers/safari.json) of WebKit to Safari versions in mdn browser compat data. Your `navigator.userAgent` string includes `AppleWebKit/605.1.15`, and the map shows this corresponds to a version of Safari between 11 and 11.1, with 11.1 released 2018-04-12.

Sorry, this is unsupportable, since `regex` relies on lookbehind, which Safari was very late to support (not until Safari 16.4, released 2023-03-27).

Closing as wontfix.
